### PR TITLE
Oslo context values generation

### PR DESCRIPTION
**Pull request by ridingwolf**

### CODE REVIEW

**Comment by ridingwolf** [comment]
`Be.Vlaanderen.Basisregisters.GrAr.Oslo` is still empty.
Files are not yet generated because the source has known typos.

Currently we could do 2 things:
- remove project again
- publish empty project

**Comment by ridingwolf** [comment]
generated example
```csharp
namespace Be.Vlaanderen.Basisregisters.GrAr.Oslo.ContextProperties.v20201127
{
    ///<summary>source: <a href="https://data.vlaanderen.be/context/adresregister.jsonld">https://data.vlaanderen.be/context/adresregister.jsonld</a></summary>
    public class Address
    {
        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Adres">https://data.vlaanderen.be/ns/adres#Adres</a></summary>
        public const string __Adres = "Adres";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#AdresseerbaarObject">https://data.vlaanderen.be/ns/adres#AdresseerbaarObject</a></summary>
        public const string __AdresseerbaarObject = "AdresseerbaarObject";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Adresuitbreiding">https://data.vlaanderen.be/ns/adres#Adresuitbreiding</a></summary>
        public const string __Adresuitbreiding = "Adresuitbreiding";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/generiek#Gemeente">https://data.vlaanderen.be/ns/generiek#Gemeente</a></summary>
        public const string __Gemeente = "Gemeente";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Gemeentenaam">https://data.vlaanderen.be/ns/adres#Gemeentenaam</a></summary>
        public const string __Gemeentenaam = "Gemeentenaam";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Ligplaats">https://data.vlaanderen.be/ns/adres#Ligplaats</a></summary>
        public const string __Ligplaats = "Ligplaats";

        ///<summary>source: <a href="http://purl.org/dc/terms/Location">http://purl.org/dc/terms/Location</a></summary>
        public const string __Plaats = "Plaats";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Postinfo">https://data.vlaanderen.be/ns/adres#Postinfo</a></summary>
        public const string __Postinfo = "Postinfo";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Standplaats">https://data.vlaanderen.be/ns/adres#Standplaats</a></summary>
        public const string __Standplaats = "Standplaats";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Straatnaam">https://data.vlaanderen.be/ns/adres#Straatnaam</a></summary>
        public const string __Straatnaam = "Straatnaam";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#busnummer">https://data.vlaanderen.be/ns/adres#busnummer</a></summary>
        public const string __Adres_busnummer = "Adres.busnummer";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#huisnummer">https://data.vlaanderen.be/ns/adres#huisnummer</a></summary>
        public const string __Adres_huisnummer = "Adres.huisnummer";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Adres.status">https://data.vlaanderen.be/ns/adres#Adres.status</a></summary>
        public const string __Adres_status = "Adres.status";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#volledigAdres">https://data.vlaanderen.be/ns/adres#volledigAdres</a></summary>
        public const string __Adresuitbreiding_volledigAdres = "Adresuitbreiding.volledigAdres";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Locatieaanduiding.aanduiding">https://data.vlaanderen.be/ns/adres#Locatieaanduiding.aanduiding</a></summary>
        public const string __Locatieaanduiding_aanduiding = "Locatieaanduiding.aanduiding";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Locatieaanduiding.type">https://data.vlaanderen.be/ns/adres#Locatieaanduiding.type</a></summary>
        public const string __Locatieaanduiding_type = "Locatieaanduiding.type";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#postcode">https://data.vlaanderen.be/ns/adres#postcode</a></summary>
        public const string __Postinfo_postcode = "Postinfo.postcode";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#postnaam">https://data.vlaanderen.be/ns/adres#postnaam</a></summary>
        public const string __Postinfo_postnaam = "Postinfo.postnaam";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#aanduiding">https://data.vlaanderen.be/ns/adres#aanduiding</a></summary>
        public const string __aanduiding = "aanduiding";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#adminUnitL1">http://www.w3.org/ns/locn#adminUnitL1</a></summary>
        public const string __administratieveEenheidNiveau1 = "administratieveEenheidNiveau1";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#adminUnitL2">http://www.w3.org/ns/locn#adminUnitL2</a></summary>
        public const string __administratieveEenheidNiveau2 = "administratieveEenheidNiveau2";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#adresIsGelegenIn">https://data.vlaanderen.be/ns/adres#adresIsGelegenIn</a></summary>
        public const string __adresIsGelegenIn = "adresIsGelegenIn";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#addressArea">http://www.w3.org/ns/locn#addressArea</a></summary>
        public const string __adresgebied = "adresgebied";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#adreslocator">https://data.vlaanderen.be/ns/adres#adreslocator</a></summary>
        public const string __adreslocator = "adreslocator";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Adresvoorstelling.busnummer">https://data.vlaanderen.be/ns/adres#Adresvoorstelling.busnummer</a></summary>
        public const string __busnummer = "busnummer";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/generiek#default">https://data.vlaanderen.be/ns/generiek#default</a></summary>
        public const string __default = "default";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#gemeentenaam">https://data.vlaanderen.be/ns/adres#gemeentenaam</a></summary>
        public const string __gemeentenaam = "gemeentenaam";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#geometry">http://www.w3.org/ns/locn#geometry</a></summary>
        public const string __geometrie = "geometrie";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#heeftGemeentenaam">https://data.vlaanderen.be/ns/adres#heeftGemeentenaam</a></summary>
        public const string __heeftGemeentenaam = "heeftGemeentenaam";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#heeftPostinfo">https://data.vlaanderen.be/ns/adres#heeftPostinfo</a></summary>
        public const string __heeftPostinfo = "heeftPostinfo";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#heeftStraatnaam">https://data.vlaanderen.be/ns/adres#heeftStraatnaam</a></summary>
        public const string __heeftStraatnaam = "heeftStraatnaam";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#homoniemToevoeging">https://data.vlaanderen.be/ns/adres#homoniemToevoeging</a></summary>
        public const string __homoniemToevoeging = "homoniemToevoeging";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Adresvoorstelling.huisnummer">https://data.vlaanderen.be/ns/adres#Adresvoorstelling.huisnummer</a></summary>
        public const string __huisnummer = "huisnummer";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#isAfgeleidVan">https://data.vlaanderen.be/ns/adres#isAfgeleidVan</a></summary>
        public const string __isAfgeleidVan = "isAfgeleidVan";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#isToegekendAan">https://data.vlaanderen.be/ns/adres#isToegekendAan</a></summary>
        public const string __isToegekendAan = "isToegekendAan";

        ///<summary>source: <a href="http://www.w3.org/ns/prov#wasAttributedTo">http://www.w3.org/ns/prov#wasAttributedTo</a></summary>
        public const string __isToegekendDoor = "isToegekendDoor";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#isVerrijktMet">https://data.vlaanderen.be/ns/adres#isVerrijktMet</a></summary>
        public const string __isVerrijktMet = "isVerrijktMet";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#land">https://data.vlaanderen.be/ns/adres#land</a></summary>
        public const string __land = "land";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#locatorDesignator">http://www.w3.org/ns/locn#locatorDesignator</a></summary>
        public const string __locatieaanduiding = "locatieaanduiding";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#locatorName">http://www.w3.org/ns/locn#locatorName</a></summary>
        public const string __locatienaam = "locatienaam";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/generiek#methode">https://data.vlaanderen.be/ns/generiek#methode</a></summary>
        public const string __methode = "methode";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#naam">https://data.vlaanderen.be/ns/adres#naam</a></summary>
        public const string __naam = "naam";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#niveau">https://data.vlaanderen.be/ns/adres#niveau</a></summary>
        public const string __niveau = "niveau";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#officieelToegekend">https://data.vlaanderen.be/ns/adres#officieelToegekend</a></summary>
        public const string __officieelToegekend = "officieelToegekend";

        ///<summary>source: <a href="http://www.w3.org/2000/01/rdf-schema#label">http://www.w3.org/2000/01/rdf-schema#label</a></summary>
        public const string __plaatsnaam = "plaatsnaam";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#positie">https://data.vlaanderen.be/ns/adres#positie</a></summary>
        public const string __positie = "positie";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#poBox">http://www.w3.org/ns/locn#poBox</a></summary>
        public const string __postbus = "postbus";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#postCode">http://www.w3.org/ns/locn#postCode</a></summary>
        public const string __postcode = "postcode";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#postName">http://www.w3.org/ns/locn#postName</a></summary>
        public const string __postnaam = "postnaam";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/generiek#specificatie">https://data.vlaanderen.be/ns/generiek#specificatie</a></summary>
        public const string __specificatie = "specificatie";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Straatnaam.status">https://data.vlaanderen.be/ns/adres#Straatnaam.status</a></summary>
        public const string __status = "status";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#thoroughfare">http://www.w3.org/ns/locn#thoroughfare</a></summary>
        public const string __straatnaam = "straatnaam";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#Locatienaam.type">https://data.vlaanderen.be/ns/adres#Locatienaam.type</a></summary>
        public const string __type = "type";

        ///<summary>source: <a href="https://data.vlaanderen.be/ns/adres#verwijstNaar">https://data.vlaanderen.be/ns/adres#verwijstNaar</a></summary>
        public const string __verwijstNaar = "verwijstNaar";

        ///<summary>source: <a href="http://www.w3.org/ns/locn#fullAddress">http://www.w3.org/ns/locn#fullAddress</a></summary>
        public const string __volledigAdres = "volledigAdres";
    }
}
```

File: paket.dependencies [hunk 1]
```diff
@@ -5,6 +5,10 @@ source https://api.nuget.org/v3/index.json
 nuget Microsoft.CSharp 4.7.0
 nuget Microsoft.Extensions.Logging 3.1.8
 
+nuget Microsoft.Extensions.Configuration 3.1.8
+nuget Microsoft.Extensions.Configuration.Json 3.1.8
+
+
```
Comment (CumpsD) [inline paket.dependencies:11]: random newline

File: tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator.csproj [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
+	<Import Project="..\..\packages\Be.Vlaanderen.Basisregisters.Build.Pipeline\Content\Be.Vlaanderen.Basisregisters.Build.Pipeline.Settings.App.props" />
+
+	<ItemGroup>
+		<Content Include="appsettings.json" CopyToOutputDirectory="Always" />
+		<Content Include="appsettings.*.json" CopyToOutputDirectory="Always" />
+	</ItemGroup>
+
+	<Import Project="..\..\.paket\Paket.Restore.targets" />
+</Project>
```
Comment (CumpsD) [inline tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator.csproj:10]: missing eof newline

File: tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/ContextInformation.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    public class ContextInformation
+    {
+        public ContextInformation(string name, Uri sourceUrl)
+        {
+            Name = name;
+            SourceUrl = sourceUrl;
+        }
+
+        public string Name { get; }
```
Comment (CumpsD) [inline tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/ContextInformation.cs:13]: props above ctor

File: src/Be.Vlaanderen.Basisregisters.GrAr.Oslo/Be.Vlaanderen.Basisregisters.GrAr.Oslo.csproj [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      <NoWarn>1701;1702;1705;1591</NoWarn>
+      <DefineConstants>TRACE;DEBUG;NETCOREAPP;NETCOREAPP3_1</DefineConstants>
+    </PropertyGroup>
+    <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+      <DocumentationFile>bin\Release\netcoreapp3.1\Be.Vlaanderen.Basisregisters.GrAr.Oslo.xml</DocumentationFile>
+      <NoWarn>1701;1702;1705;1591</NoWarn>
+    </PropertyGroup>
+
+    <Import Project="..\..\.paket\Paket.Restore.targets" />
+</Project>
```
Comment (CumpsD) [inline src/Be.Vlaanderen.Basisregisters.GrAr.Oslo/Be.Vlaanderen.Basisregisters.GrAr.Oslo.csproj:15]: missing eof newline

File: src/Be.Vlaanderen.Basisregisters.GrAr.Oslo/paket.template [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+authors Basisregisters Vlaanderen
+owners agentschap Informatie Vlaanderen <https://vlaanderen.be/informatie-vlaanderen>
+copyright Copyright (c) agentschap Informatie Vlaanderen
+requireLicenseAcceptance false
+projectUrl https://github.com/Informatievlaanderen/grar-common
+iconUrl https://raw.githubusercontent.com/Informatievlaanderen/build-pipeline/master/logo.png
+licenseUrl https://opensource.org/licenses/EUPL-1.2
+
+description Contains common oslo classes from GrAr.
```
Comment (CumpsD) [inline src/Be.Vlaanderen.Basisregisters.GrAr.Oslo/paket.template:15]: uppercase OSLO

File: tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/ContextPropertiesFileBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            var objectToParse = jObject.ContainsKey("@context")
+                ? jObject["@context"]
+                : jObject;
+
+            foreach (var pair in objectToParse ?? throw new ApplicationException($"Context is not found in: {jObject}"))
+            {
+                if (pair is JProperty property)
+                {
+                    var reference = property.Value.Count() > 1
+                        ? property.Value["@id"]?.ToString() ?? ""
```
Comment (CumpsD) [inline tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/ContextPropertiesFileBuilder.cs:50]: `""` --> string.Empty

File: tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/ContextPropertiesFileBuilder.cs [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+            }
+
+            builder.AppendLine($"{Indent(1)}}}");
+            builder.AppendLine("}");
+
+            return builder.ToString();
+        }
+
+        private static string Indent(int amount)
+            => "".PadLeft(4 * amount);
```
Comment (CumpsD) [inline tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/ContextPropertiesFileBuilder.cs:93]: new string('', 4 * amount)

File: tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/ContextPropertyDefinition.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        public ContextPropertyDefinition(
+            string key,
+            string reference)
+        {
+            Name = $"__{key.Replace('.', '_')}";
+            Value = key;
+            Reference = reference;
+        }
+
+        public string Name { get; }
```
Comment (CumpsD) [inline tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/ContextPropertyDefinition.cs:14]: props above ctor

File: tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/Program.cs [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Configuration;
+
+    class Program
```
Comment (CumpsD) [inline tools/Be.Vlaanderen.Basisregisters.Oslo.ContextProperties.Generator/Program.cs:11]: public class Program

**Comment by ridingwolf** [comment]
`Be.Vlaanderen.Basisregisters.GrAr.Oslo` needs to be created on nuget before triggering the build

**Pull request merged**